File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.go
func longestConsecutive(nums []int) int {
    m := make(map[int]int)
    if len(nums) == 0{
        return 0
    }

    for _, v := range nums{
        m[v] += 1
    }

    ans := 1

    for _, v := range nums{
        if _, ok := m[v-1]; ok{
            continue
        }

        t := 1
        for i:= v+1;;i++{
            if _, ok := m[i]; ok{
                t += 1
                ans = max(ans, t)
                continue
            }
            break
        }
        
    }
    return ans

}

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        prefix = 1
        suffix = 1
        ans = float('-inf') 

        for i in range(len(nums)):
            vp = nums[i]
            vs = nums[len(nums)-i-1]

            prefix *= vp 
            suffix *= vs

            ans = max([prefix,suffix,ans])

            if vp == 0:
                prefix = 1
            if vs == 0:
                suffix = 1

        return ans
            
            

                

        